### DISCUSSION TITLE
Awake_start and awake_stop scripts for turning SentryMode on and off for archiving

### DISCUSSION THREAD

**Discussion by jjermany** [marcone/teslausb#902] [Ideas]
[awake_start_MODIFIED.txt](https://github.com/user-attachments/files/17905777/awake_start_MODIFIED.txt)
[awake_stop_MODIFIED.txt](https://github.com/user-attachments/files/17905780/awake_stop_MODIFIED.txt)

Ill need someone to integrate this logic into our existing awake start and awake stop scripts. It uses TeslaFi (if you have an account). I added a variable to our config file:

# TESLAFI WAKE MODE: Uncomment line below to choose stream or sentry
export TESLAFI_WAKE_MODE=sentry  # or "stream"

**Comment by miles267** [comment]
@jjermany thanks for this.  Look forward to the PR for START and STOP being integrated into the Teslausb code.

**Reply by jjermany** [reply]
No problem at all. I absolutely love this project and just happy to help.

**Reply by miles267** [reply]
@jjermany until this PR is merged into the main code, would mind providing some brief instructions on how to implement your scripts/process above?  I'd like to use it for both of our Tesla cars.  I'm already a TeslaFi user but the limited wake command support has caused my Teslausb syncs to no longer complete unless I leave sentry mode ON at home.  In which case it also results in more video files.

For example, do I repace the awake_start and awake_stop files in /root/bin with the two files above?  If I copy them into the /root/bin folder (without the *_modified.txt, of course), do they need to be chmod'd?  Do I then just also add into the .conf file the export TESLAFI_WAKE_MODE=sentry entry?  Do I need to re-run the teslausb upgrade script after that entry is made to the .conf file OR just reboot the teslausb system?

**Comment by jjermany** [comment]
@miles267 You will replace the awake_start and awake_stop files with the
two above. You will also want to add the  TESLAFI_WAKE_MODE=sentry to the
conf file. This should get the desired result. Let me know if you have any
issues after making these changes.

**Reply by miles267** [reply]
@jjermany thanks.  I'm online about to do this now 👍 Could you please tell me:

1.) once I copy those 2 scripts to /root/bin, do they need to be chmod'ded in a certain way?  I'm asking because I'd prefer to keep both backups of the original files and place the new ones in that folder for use.

2.) after I've added the TESLAFI_WAKE_MODE=sentry entry to the .conf file, is it necessary to re-run the teslausb upgrade script?  Or is rebooting the pi enough?

**Comment by jjermany** [comment]
You should not need to chmod those files. I did not have to. You are just
replacing the current wake scripts with the new ones. No chmod is necessary
here.

Unfortunately, once you run the teslausb upgrade script, it will wipe out
the wake files to the default ones. Rebooting works great in this case.

**Reply by miles267** [reply]
Thanks again @jjermany.  Sorry, hopefully one last question.  Do I want the "Exclude Home" box to be CHECKED or UNCHECKED in the car's Sentry Mode config UI?

**Reply by jjermany** [reply]
You can have the car exclude your home. It will automatically pull your TeslaFi credentials and will turn on sentry mode. It will turn off after the sync is complete.

**Reply by miles267** [reply]
Interesting.  I've done this now in both cars and it is not turning Sentry mode on/off.  I'm editing the /root/teslausb_setup_variables.conf file (but not re-running the teslausb upgrade procedure).  Should I be editing a different file to include this entry?

export TESLAFI_WAKE_MODE=sentry

**Reply by marcone** [reply]
> Interesting. I've done this now in both cars and it is not turning Sentry mode on/off

This is likely because (as I mentioned in my review comments on the PR), you also need to set TESLA_WAKE_MODE to "sentry", because of the way this new code was intertwined with the existing code.

**Reply by miles267** [reply]
@marcone actually that didn't address the issue either.  Yet I can confirm that I can manually toggle sentry mode ON and OFF via TeslaFi and the car responds accordingly.  Hopefully this testing helps with the PR.

**Comment by jjermany** [comment]
Can you restart your pi and send us the archive loop log?

**Reply by miles267** [reply]
> Can you restart your pi and send us the archive loop log?
> […](#)
> On Sat, Feb 8, 2025, 5:19 PM miles267 ***@***.***> wrote: @marcone <https://github.com/marcone> actually that didn't address the issue either. Yet I can confirm that I can manually toggle sentry mode ON and OFF via TeslaFi and the car responds accordingly. Hopefully this testing helps with the PR. — Reply to this email directly, view it on GitHub <[#902 (reply in thread)](https://github.com/marcone/teslausb/discussions/902#discussioncomment-12105817)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A546GXPZM2HGSOD76VJGCM32O2GH3AVCNFSM6AAAAABSOL25R6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMJQGU4DCNY> . You are receiving this because you were mentioned.Message ID: ***@***.***>

@jjermany Attached per your request.  This is after:
1.) replacing awake_start and awake_stop in /root/bin folder with the modified versions
2.) adding my TeslaFI API in teslausb .conf file
3.) adding the following two entries to the teslausb .conf file:
export TESLA_WAKE_MODE=sentry
export TESLAFI_WAKE_MODE=sentry
4.) confirming I'm able to successfully toggle Sentry mode ON and OFF from my TeslaFi web dashboard
[archiveloop.log](https://github.com/user-attachments/files/18725707/archiveloop.log)

**Reply by marcone** [reply]
Your log shows it printing "Starting background task to keep car awake" every time it archives, which is the message from the original TeslaUSB version of awake_start. That message is different in the version of awake_start from PR https://github.com/marcone/teslausb/pull/928 and does not appear in the [awake_start_MODIFIED.txt](https://github.com/user-attachments/files/17905777/awake_start_MODIFIED.txt) file at the top of this discussion, so to me it looks like you didn't actually replace awake_start.

**Reply by marcone** [reply]
(and additionally, that message is only printed when TESLA_WAKE_MODE is either not set, or set to "stream", so it looks like you also didn't set the right values in the .conf file)

**Reply by miles267** [reply]
@marcone confirming I did replace both the awake_start and awake_stop files. I actually deleted the original ones from /root/bin, copying the new/modified ones in their place.

However I had previously set TESLA_WAKE_MODE=sentry (not =stream) and TESLAFI_WAKE_MODE=sentry in the .conf file.  Here is the archiveloop file with the awake_start and awake_stop replaced and only TESLAFI_WAKE_MODE=sentry.
[archiveloop2.log](https://github.com/user-attachments/files/18726725/archiveloop2.log)

**Reply by marcone** [reply]
> @marcone confirming I did replace both the awake_start and awake_stop files. I actually deleted the original ones from /root/bin, copying the new/modified ones in their place.

The archiveloop log shows otherwise though. The message that it is logging does not appear in either the modified awake_start attached above or the one from the PR. Therefore, you must still be using the original one.
 
> However I had previously set TESLA_WAKE_MODE=sentry (not =stream) and TESLAFI_WAKE_MODE=sentry in the .conf file. Here is the archiveloop file with the awake_start and awake_stop replaced and only TESLAFI_WAKE_MODE=sentry. [archiveloop2.log](https://github.com/user-attachments/files/18726725/archiveloop2.log)

What I was saying is that TESLA_WAKE_MODE needs to be set to "sentry" (at least for the awake_start from the PR), but the archiveloop log shows it was either set to "stream", or not set at all.

Edit: your new archiveloop log appears to show that awake_start is not being called, or failing without logging a message.
You can try running `journalctl -u teslausb.service` to see if that shows any additional error messages. Also make sure the replaced awake_start script is executable.

**Reply by miles267** [reply]
@marcone do I need to be doing something other than replacing awake_start and awake_stop in /root/bin folder with the new files from the PR? That’s what I’ve done yet it doesn’t use them. And I’m not re-running the Teslausb upgrade script that would overwrite them either.

**Reply by marcone** [reply]
try just running them manually and see if it complains about anything: `sudo /root/bin/awake_start`

**Reply by miles267** [reply]
@marcone when I attempt to run via: sudo /root/bin/awake_start, it says command not found.  However if I run sudo -i and check the folder, it shows the folder is there.  But perhaps a chmod command needs to be issued to make its permissions the same as the other scripts as shown here?

EDIT: I chmod +x to each script and they'll now run as sudo /root/bin/awake_start or awake_stop.  Doesn't appear it's toggling sentry mode on and off though.

![Screenshot 2025-02-10 074519](https://github.com/user-attachments/assets/ca047977-6698-4857-a6b5-4ee3e54d9fc0)

**Reply by marcone** [reply]
> EDIT: I chmod +x to each script and they'll now run as sudo /root/bin/awake_start or awake_stop. Doesn't appear it's toggling sentry mode on and off though.

That could be because the relevant variables aren't set when called like this. It should work work when called from TeslaUSB, or you can verify like this:
```
sudo -i
source /root/bin/envsetup.sh
/root/bin/awake_start
/root/bin/awake_stop
```

**Comment by miles267** [comment]
Confirming I was able to get this to work using the two files: awake_start_MODIFIED.txt and awake_stop_MODIFIED.txt with TeslaFi.  However it appears the awake and stop scripts in the PR are different (filesize, etc.) and wouldn't work when I placed them in /root/bin.  @jjermany any progress on updating this PR so the code can be merged officially into teslausb main branch?

**Reply by miles267** [reply]
@jjermany I noticed today my Teslausb connected to wifi and began syncing as soon as I pulled into the garage, even before the car was put into PARK.  This resulted in the TeslaFi attempt to enable sentry mode being rejected because the car wasn't yet parked.  Is there any way to perhaps add a short delay to this (or a variable to allow for it), while you're compiling this PR?

02/11/2025 6:45 PM, set_sentry_mode, 123.123.123.123, car could not execute command: not_parked

**Reply by marcone** [reply]
The problem with adding a "short delay" is that the car could go to sleep during this delay, or that sometimes it is not long enough because you took a little longer than usual to park the car. A better solution might be to have awake_start retry putting the car in sentry mode in the background. That would end up eating into the 500 commands/month allowance though, so there really is no perfect solution here it seems.

**Reply by miles267** [reply]
> The problem with adding a "short delay" is that the car could go to sleep during this delay, or that sometimes it is not long enough because you took a little longer than usual to park the car. A better solution might be to have awake_start retry putting the car in sentry mode in the background. That would end up eating into the 500 commands/month allowance though, so there really is no perfect solution here it seems.

@marcone Understood.  That logic does make sense.  Based on my early testing, with this in mind, I'm finding that TeslaFi Sentry Mode is enabled about 50% of the time.  As a matter of timing of events (pulling into driveway, AP signal being detected, initiating Teslausb sync process and managing to get the car into PARK mode).  It may very well be that my Raspberry Pi is connecting to my wireless AP more quickly than I'm able to pull into the garage and put the car into PARK.

@jjermany what is the status of this PR?  I just have to remember not to run the teslausb upgrade script, in the interim, else it reverts this TeslaFi Sentry Mode awake_start and awake_stop modification.

**Reply by jjermany** [reply]
@miles267 @marcone I just finished updating those awake_start and awake_stop scripts. Works beautifully. It checks the TeslaFi telemetry data for current sentry status. This is essential for people who do NOT exclude their home from sentry mode. The script will not turn off sentry if it reads that sentry is already enabled, otherwise, it will turn it back off. I sent the code for PR #936 . @marcone  please ping me if you need anything else from me. Again, I'm fairly new to GitHub so very sorry in advance if I didn't submit it correctly.
